rcjaRegistration/events/views.py

```diff
@@ -234,3 +236,103 @@ def delete(self, request, teamID=None, attendeeID=None, eventID=None):
         # Delete team
         eventAttendance.delete()
         return HttpResponse(status=204)
+
+def getEventsForSummary(state, year):
+    """ Create list of event dictionaries of all events in state and year """
+    eventList = Event.objects.filter(state = state, year = year)
+
+    # Find information for events
+    events = []
+    for event in eventList:
+        eventDict = {}
+        eventDict["name"] = event.name
+
+        if event.startDate==event.endDate:
+            eventDict["date"] = event.startDate
+        else:
+            eventDict["date"] = f"{event.startDate.strftime('%d/%m/%Y')} - {event.endDate.strftime('%d/%m/%Y')}"
+
+
+        attendances = BaseEventAttendance.objects.filter(event=event)
+        teamNumber = 0
+        studentNumber = 0
+        teacherNumber = 0
+        maleNumber = 0
+        femaleNumber = 0
+        otherNumber = 0
+        
+        for attendance in attendances:
+            # Find type
+            if attendance.eventAttendanceType() == "team": # Team
+                teamNumber += 1
+                students = Student.objects.filter(team=attendance.childObject())
+                for student in students:
+                    studentNumber += 1
+                    if student.gender == "male":
+                        maleNumber += 1
+                    elif student.gender == "female":
+                        femaleNumber += 1
+                    else:
+                        otherNumber += 1
+            elif attendance.eventAttendanceType() == "workshopattendee": # Workshop
+                if attendance.attendeeType == "student":
```

Comment:
It appears this doesn't work when attendees exist, I think your data might have them as 0's so it isn't running this part of the code.
![image](https://github.com/user-attachments/assets/65c761b8-66a5-4935-9540-cb426c5c0f8b)

You can use `attendance.childObject()` to get the WorkshopAttendee object for it which will give you what you need

However, it might be better to move the attendance logic down to the `if event.eventType == "competition":` since then you can grab directly from `WorkshopAttendee` and `Team` which should avoid the need for the `if attendance.eventAttendanceType() == "team"` logic

Reply:
> I think your data might have them as 0's so it isn't running this part of the code.

Along with this, it'd be good to get some tests that can cover this part too, if you look at the codecov report you can find what parts are missing test coverage at the moment. 
Let me know if you need help figuring out how to write them.